cascade-merge/action.yml

```diff
@@ -0,0 +1,162 @@
+            TARGET_VERSIONS="$TARGET_VERSIONS $VERSION"
+          fi
+        done
+        
+        NEW_PR_URLS=""
+        
+        # Loop through all target versions
+        for TARGET_VERSION in $TARGET_VERSIONS; do
+          if [[ $TARGET_VERSION == "${{ inputs.default-branch }}" ]]; then
+            TARGET_BRANCH="${{ inputs.default-branch }}"
+          else
+            TARGET_BRANCH="$BACKPORT_TARGET_BRANCH_PATTERN$TARGET_VERSION"
+          fi
+        
+          BRANCH_NAME="$PREFIX-${ORIGINAL_BRANCH}-to-$TARGET_BRANCH"
+
+          # Get the PR number of the existing PR (if any) that has the same base and head
+          PR_NUMBER=$(gh pr list --base $TARGET_BRANCH --head $BRANCH_NAME --json number --jq '.[0].number')
+        
+          # If a PR number exists, close that PR and delete the branch        
+          if [ "$PR_NUMBER" -eq "$PR_NUMBER" ] 2>/dev/null; then
+            echo_green "closing old PR $PR_NUMBER"
+            gh pr close $PR_NUMBER --delete-branch
+            echo_green "old PR $PR_NUMBER was closed"
+          fi
+        
+          # Create a new branch, merge the original branch into it, and push it
+          git checkout -b $BRANCH_NAME
+          git merge $ORIGINAL_BRANCH
+          git push -f origin $BRANCH_NAME
+          
+          # Attempt to create a new PR and store the return status
+          echo_green "attempting to create a pull request to cascade merge from $ORIGINAL_BRANCH to $TARGET_BRANCH"
+          NEW_PR_URL=$(gh pr create --title "Cascade merge from $ORIGINAL_BRANCH to $TARGET_BRANCH" --body "Automated cascade merge" --base $TARGET_BRANCH --head $BRANCH_NAME --repo ${{ github.repository }} --draft) || PR_CREATION_STATUS=$?
+          if [[ $PR_CREATION_STATUS -ne 0 ]]; then
+              # If PR creation failed, print a message and delete the branch both locally and remotely
+              echo_green "PR creation failed. Deleting branch."
+              git checkout $ORIGINAL_BRANCH
+              git branch -d $BRANCH_NAME
+              git push origin --delete $BRANCH_NAME
+          else
+              echo_green "PR is created $NEW_PR_URL"
+              # If PR creation succeeded, make the PR ready for review
+              gh pr ready "${NEW_PR_URL##*/}"
+              echo_green "PR is marked as ready $NEW_PR_URL"
+          fi
+        
+          NEW_PR_URLS="$NEW_PR_URLS $NEW_PR_URL"
+        
+          # If a PR number exists, add a comment to it pointing to the new PR
+          if [ "$PR_NUMBER" -eq "$PR_NUMBER" ] 2>/dev/null; then
+            gh pr comment $PR_NUMBER --body "This PR is continued in $NEW_PR_URL"
+          fi
+        
+          # Checkout back to the original branch
+          git checkout $ORIGINAL_BRANCH
+        done
+
+        # Pass the new PR URLs to the next steps in the job
+        echo "new_pr_urls=$NEW_PR_URLS" >> $GITHUB_ENV
+      shell: bash
+      env:
+        GH_TOKEN: ${{ inputs.gh-token }}
+
+    - name: Merge PRs
```

Comment:
It will be merged every time !

actually, I'm not sure we should ever automerge. 
At least we should make sure the CI pass correctly

Reply:
we merge directly. no PR is created